src/Storages/Distributed/DistributedSettings.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <Core/Defines.h>
+#include <Core/BaseSettings.h>
+
+
+namespace Poco::Util
+{
+    class AbstractConfiguration;
+}
+
+
+namespace DB
+{
+class ASTStorage;
+
+#define LIST_OF_DISTRIBUTED_SETTINGS(M) \
+    M(Bool, fsync_after_insert, false, "Do fsync for every inserted. Will decreases performance of inserts (only for async INSERT, i.e. insert_distributed_sync=false)", 0) \
+    M(Bool, fsync_tmp_directory, false, "Do fsync for temporary directory (that is used for async INSERT only) after all part operations (writes, renames, etc.).", 0) \
```

Comment:
I have mixed feeling here... From one side i really like the idea of having Distributed-table related settings stored with table, from the other - we have already dozens of Distributed table settings stored as user-level settings

Maybe we can start moving there here, and some common pattern like "user settings will be used unless table settings redifine it". 

Maybe it's worth to add a test with ALTER TABLE ... MODIFY SETTINGS ...?

Reply:
>we have already dozens of Distributed table settings stored as user-level settings

Some of them will be migrated to SETTINGS (there is TODO for this)

>Maybe we can start moving them here, and use some common pattern like "user settings will be used unless table settings redifine it"?

Yes, but separately, since it is not related to fsync.

>Maybe it's worth to add a test with ALTER TABLE ... MODIFY SETTINGS ...?

Right now `ALTER MODIFY SETTING` is not support

Reply:
:+1: